src/map.c

```diff
@@ -620,6 +620,20 @@ int wally_keypath_xonly_public_key_verify(const unsigned char *key, size_t key_l
     return WALLY_OK;
 }
 
+int wally_merkle_path_xonly_public_key_verify(const unsigned char *key, size_t key_len,
+                                          const unsigned char *val, size_t val_len)
+{
+    struct ext_key extkey;
+
+    if (key_len != EC_XONLY_PUBLIC_KEY_LEN ||
+        keypath_key_verify(key, key_len, &extkey) != WALLY_OK ||
+        extkey.version ||
+        !val ||
+        val_len % SHA256_LEN != 0)
```

Comment:
`val_len` of zero should be failed?

also nit: please bunch up the conditions.